proxy/src/auth/credentials.rs

```diff
@@ -174,6 +174,10 @@ impl ComputeUserInfoMaybeEndpoint {
 }
 
 pub(crate) fn check_peer_addr_is_in_list(peer_addr: &IpAddr, ip_list: &[IpPattern]) -> bool {
+    if ip_list.len() == 1 && ip_list[0] == IpPattern::Single(IpAddr::from([127, 0, 0, 1])) {
+        // This is a special pattern which disables all access if the only one is 127.0.0.1.
+        return true;
+    }
```

Comment:
I think we want this to return false when it's disabled.

Secondarily, I think rather than hardcoding 127.0.0.1 as a sentinel value, we can instead deny all requests from loopback, we will simply never support that anyway (this would mean that a process from proxy pod is requesting into the proxy process)
```suggestion
    if peer_addr.is_loopback() {
        return false;
    }
```